test-deployment/BUILD

```diff
@@ -0,0 +1,21 @@
+load("@graknlabs_build_tools//checkstyle:rules.bzl", "checkstyle_test")
+
+java_test(
+    name = "test-deployment",
+    test_class = "TestDeployment",
+    srcs = ["TestDeployment.java"],
+    deps = [
+        "//common:common",
+        "@graknlabs_client_java//:client-java",
+        "@graknlabs_graql//java:graql",
+        "//server:server", # NEEDS TO GO
+        "@graknlabs_grakn_core//concept"
+    ]
+)
+
+checkstyle_test(
+    name = "checkstyle",
+    targets = [
+        ":test-deployment"
+    ],
+)
```

Comment:
I think this package should be call `//test/deployment` instead of `//test-deployment`, then `//test` can also contain `../assembly` from `./circleci` as commented above.

Reply:
Can we do it on a follow up PR? In that PR we can also move `test-integration` and `test-end-to-end` so everything is consistent:
```
test/
    integration
    end-to-end
    deployment
```

Reply:
Well, I think we can make this first PR to be correct without stretching the scope, and still fix the whole thing in the next PR. I.e.:
1. Move `//test/deployment` in this PR.
2. Move  the old tests (which we may remove soon) to `//test/end-to-end` and `//test/integration` in the next PR

Reply:
Done in 1e1eb2e79b68e3d3952a045e5c6f9752ae37815d.

The structure becomes like this:
```
test/
    assembly
    deployment
    common # contains 'GraknApplicationTest' which is shared by both assembly and deployment
```

Reply:
There will be a follow-up PR for this issue: https://github.com/graknlabs/grakn/issues/5043